doc/feature/cppPlugins.md

```diff
@@ -0,0 +1,159 @@
+C++ Plugin API for CPU compressors  {#data}
+============
+
+This document specifies a modern API for the in-memory C plugin API for
+lossless byte data compressors. It will not replace the 2D image
+compressors and CPU-GPU transfer plugins.
+
+## Requirements
+
+* Support all features used by current lossless byte compressors
+    * Lossless compression of binary blobs
+    * Multiple compressors per plugin DSO
```

Comment:
DSO refers to dynamic linking as far as I understand. Are plugins really going to be loaded at runtime?